circle.yml

```diff
@@ -28,4 +28,23 @@ test:
     - cd $SRCDIR/cover; make
     - cd $SRCDIR/socks; make
     - cd $SRCDIR/runner; make
+    - cd $SRCDIR/build; make
 
+deployment:
+  snapshot:
+    branch: master
+    commands:
+      - docker login -e "$DOCKER_REGISTRY_EMAIL" -u "$DOCKER_REGISTRY_USER" -p "$DOCKER_REGISTRY_USER"
+        # Base the image tag on the version of the base-image used in the
+        # dockerfile, so "FROM golang:1.8.0-stretch" => "golang-1.8.0-stretch"
```

Comment:
I *think* it would be clearer to say:

```yaml
# Tag the built images with something derived from the base images in their respective Dockerfiles. So "FROM golang:1.8.0-stretch" as a base image would lead to a tag of "golang-1.8.0-stretch"
```

That said, I don't quite get which part of the code below does that.